test/integration/controlplane/apiserver_identity_test.go

```diff
@@ -98,7 +98,9 @@ func TestCreateLeaseOnStart(t *testing.T) {
 		}
 
 		lease := leases.Items[0]
-		if lease.Name != expectedAPIServerIdentity(t, hostname) {
+		// we skip last 4 chars in lease.Name for validation since these are a part of randomly generated suffix
```

Comment:
we should not conflate both changes, I think we should leave the randomization as a seperate pr

Reply:
created https://github.com/kubernetes/kubernetes/pull/118958 for this, but I need this change in this PR in order for the integration test to work.